### PR TITLE
Feat (Auth): Interface Implementation for fetchCurrentDevice

### PR DESCRIPTION
**Pull request by hahnandrew**
- **feat: interface implementation for fetchCurrentDevice**
- **feat: added categories for fetchCurrentDevice**

*Description of changes:*

Added interface implementation for fetchCurrentDevice API.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_core/lib/src/category/amplify_auth_category.dart [hunk 1]
```diff
@@ -1355,6 +1355,37 @@ class AuthCategory extends AmplifyCategory<AuthPluginInterface> {
         () => defaultPlugin.rememberDevice(),
       );
 
+  /// {@template amplify_core.amplify_auth_category.fetch_current_device}
+  /// Retrieves the current device.
+  ///
+  /// For more information about device tracking, see the
+  /// [Amplify docs](https://docs.amplify.aws/lib/auth/device_features/q/platform/flutter/).
```
Comment (khatruong2009) [inline packages/amplify_core/lib/src/category/amplify_auth_category.dart:1362]: Link this to the [gen 2 docs](https://docs.amplify.aws/flutter/build-a-backend/auth/manage-users/manage-devices/). I guess we'll also have to change these across the board later.

**Review by khatruong2009 — APPROVED** [review]

File: packages/amplify_core/lib/src/category/amplify_auth_category.dart (hunk 1 above)
Comment (hahnandrew) [reply packages/amplify_core/lib/src/category/amplify_auth_category.dart]: [done](https://github.com/aws-amplify/amplify-flutter/pull/5070/commits/b52049328a3a23bc02aae109ec21c6cdc455f73f)

File: packages/amplify_core/doc/lib/auth.dart [hunk 2]
```diff
@@ -483,6 +483,17 @@ Future<void> forgetSpecificDevice(AuthDevice myDevice) async {
 }
 // #enddocregion forget-specific-device
 
+// #docregion fetch-current-device
+Future<void> fetchCurrentUserDevice() async {
+  try {
+    final device = await Amplify.Auth.fetchCurrentDevice();
+    safePrint('Device: $device');
+  } on AuthException catch (e) {
+    safePrint('Get current device failed with error: $e');
```
Comment (tyllark) [inline packages/amplify_core/doc/lib/auth.dart:492]: Should this say fetch current device?

File: packages/amplify_core/lib/src/category/amplify_auth_category.dart [hunk 3]
```diff
@@ -1355,6 +1355,37 @@ class AuthCategory extends AmplifyCategory<AuthPluginInterface> {
...
+  /// ```
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="fetch-current-device"?>
+  /// ```dart
+  /// Future<AuthDevice> getCurrentUserDevice() async {
+  ///   try {
+  ///     final device = await Amplify.Auth.fetchCurrentDevice();
+  ///     safePrint('Device: $device');
+  ///   } on AuthException catch (e) {
+  ///     safePrint('Get current device failed with error: $e');
```
Comment (tyllark) [inline packages/amplify_core/lib/src/category/amplify_auth_category.dart:1379]: Should this say fetch current device?

**Review by tyllark — APPROVED** [review]

File: packages/amplify_core/lib/src/category/amplify_auth_category.dart (hunk 3 above)
Comment (hahnandrew) [reply packages/amplify_core/lib/src/category/amplify_auth_category.dart]: [fixed](https://github.com/aws-amplify/amplify-flutter/pull/5070/commits/045b5b71a1b0b9760a9bb0d8102aa136f682948e)

File: packages/amplify_core/doc/lib/auth.dart (hunk 2 above)
Comment (hahnandrew) [reply packages/amplify_core/doc/lib/auth.dart]: [fixed](https://github.com/aws-amplify/amplify-flutter/pull/5070/commits/045b5b71a1b0b9760a9bb0d8102aa136f682948e)

**Pull request merged**